credentials/src/test/kotlin/web5/sdk/credentials/PresentationExchangeTest.kt

```diff
@@ -427,4 +430,152 @@ class PresentationExchangeTest {
+
+      assertEquals( 2, selectedCreds.size)
+      assertEquals( listOf(vcJwt1, vcJwt2), selectedCreds)
+    }
+
+    @Test
+    fun `selects 2 correct credential out of 3`() {
+      val pd = jsonMapper.readValue(
+        readPd("src/test/resources/pd_filter_array_single_path.json"),
+        PresentationDefinitionV2::class.java
+      )
+
+      val vc1 = VerifiableCredential.create(
+        type = "StreetCred",
+        issuer = issuerDid.uri,
+        subject = holderDid.uri,
+        data = StreetCredibility(localRespect = "high", legit = true)
+      )
+      val vcJwt1 = vc1.sign(issuerDid)
+
+      val vc2 = VerifiableCredential.create(
+        type = "StreetCred",
+        issuer = issuerDid.uri,
+        subject = holderDid.uri,
+        data = StreetCredibility(localRespect = "high", legit = true)
+      )
+
+      val vcJwt2 = vc2.sign(issuerDid)
+
+      val vc3 = VerifiableCredential.create(
+        type = "DateOfBirth",
+        issuer = issuerDid.uri,
+        subject = holderDid.uri,
+        data = DateOfBirth(dateOfBirth = "1-1-1111")
+      )
+
+      val vcJwt3 = vc3.sign(issuerDid)
+
+      val selectedCreds = PresentationExchange.selectCredentials(listOf(vcJwt1, vcJwt2, vcJwt3), pd)
+
+      assertEquals( 2, selectedCreds.size)
+      assertEquals( listOf(vcJwt1, vcJwt2), selectedCreds)
+    }
+
+    @Test
+    fun `selects 2 correct credential with two input descriptors`() {
+      val pd = jsonMapper.readValue(
+        readPd("src/test/resources/pd_filter_array_multiple_input_descriptors.json"),
+        PresentationDefinitionV2::class.java
+      )
+
+      val vc1 = VerifiableCredential.create(
+        type = "DateOfBirthSSN",
+        issuer = issuerDid.uri,
+        subject = holderDid.uri,
+        data = DateOfBirthSSN(dateOfBirth = "1999-01-01", ssn = "456-123-123")
+      )
+      val vcJwt1 = vc1.sign(issuerDid)
+
+      val vc2 = VerifiableCredential.create(
+        type = "DateOfBirthSSN",
+        issuer = issuerDid.uri,
+        subject = holderDid.uri,
+        data = DateOfBirth(dateOfBirth = "1999-01-01")
+      )
+      val vcJwt2 = vc2.sign(issuerDid)
+
+      val selectedCreds = PresentationExchange.selectCredentials(listOf(vcJwt1, vcJwt2), pd)
+
+      assertEquals( 2, selectedCreds.size)
+      assertEquals( listOf(vcJwt1, vcJwt2), selectedCreds)
+    }
+  }
 }
+
+
+class Web5TestVectorsPresentationExchangeTest {
+  data class SelectCredTestInput(
+    val presentationDefinition: PresentationDefinitionV2,
+    val credentialJwts: List<String>
+  )
+
+  data class SelectCredTestOutput(
+    val selectedCredentials: List<String>
+  )
+
+  private val mapper = jacksonObjectMapper()
+  @Test
+  fun select_credentials() {
+    val typeRef = object : TypeReference<TestVectors<SelectCredTestInput, SelectCredTestOutput>>() {}
+    val testVectors = mapper.readValue(File("../test-vectors/presentation_exchange/select_credentials.json"), typeRef)
```

Comment:
Should there be tests for the error cases ? Some examples I'm thinking about:
1. `UnsupportedOperationException` is thrown
2. When `inputDescriptorToVcMap.size != presentationDefinition.inputDescriptors.size`
3. When `JsonPathParseException` is thrown.

Reply:
+1 I think some of the tests above (multiple input descriptors, filtering array vs filtering on value) could be included in the vectors as well

Reply:
oh yes many more test vectors coming soon (tm), 
I was doing this one simple success as a steel thread for the overall web5-kt + web5-js check marks in web5-sdk-development